app/controllers/api/v1/products_controller.rb

```diff
@@ -2,11 +2,13 @@ class Api::V1::ProductsController < Api::ApiController
   before_action :set_product, only: [:show, :update, :destroy]
 
   def index
-    @products = Product.all
+    if @current_user.present?
+      @products = @current_user.products.order(created_at: :desc)
+    end
```

Comment:
You don't need to this check. Remember that if `@current_user` isn't available, a 401 will be raised.
```suggestion
    @products = @current_user.products.order(created_at: :desc)
```

Reply:
@Rubncal04 Still missing this change